src/components/global/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,60 @@
+import { ReactEventHandler, RefObject, useEffect, useRef } from "react";
```

Comment:
just a note: on deriv.app we don’t extract all elements from react. we use import React from 'react', and then React.useEffect, React.useRef etc..  
but it’s ok if it is a current practice for this project